.github/scripts/check-deploy-permissions.js

```diff
@@ -0,0 +1,63 @@
+// @ts-check
+
+/**
+ * @typedef {Schemas['webhook-pull-request-opened']} PullRequest
+ */
+
+/**
+ * Check if the current GitHub event should trigger a deployment
+ * @param {object} params
+ * @param {object} params.core - GitHub Actions core utilities
+ * @param {object} params.context - GitHub Actions context
+ */
+module.exports = async ({ core, context }) => {
+
+  // check if author is authorized
+  if (context.eventName === 'pull_request') {
+    const event = /** @type {PullRequest} */ (context.payload);
+    const authorAssociation = event.pull_request.author_association;
+
+    if (!isAllowedAuthor(authorAssociation)) {

```

Comment:
Can we trigger deployments on manual `pull_request_review` comments from maintainers like we do for extension? Maybe as a follow-up.